docs/source/networks.rst

```diff
@@ -0,0 +1,311 @@
+assigns a random boolean value to a property named `exploratory`. When this
+property is True, it means that the current network is part of the
+exploratory data subset.
+
+Creating a network
+------------------
+
+In addition to the available networks, it's fairly simple to create a custom
+network, in case an experiment design calls for different node
+interconnections. To create one, we can subclass from the Network model:
+
+::
+
+    from dallinger.models import Network
+    from dallinger.nodes import Source
+
+
+    class Ring(Network):
+
+        __mapper_args__ = {"polymorphic_identity": "ring"}
+
+        def add_node(self, node):
+            other_nodes = [n for n in self.nodes() if n.id != node.id]
+
+            if isinstance(node, Source):
+                raise Exception(
+                    "Ring network cannot contain sources."
+                )
+
+            if other_nodes:
+                parent = max(other_nodes, key=attrgetter('creation_time'))
+                parent.connect(whom=node)
+
+                if len(self.nodes) == self.max_size:
+                    parent = min(other_nodes, key=attrgetter('creation_time'))
+                    node.connect(whom=parent)
+                   
+In the above example, we create a simple `ring` network, where each node is
+connected in chain to the next one, until we get to the last one, which is
+connected back to the first, making a full circle (thus, the ring name).
+
+Ring is a subclass of `dallinger.models.Network`, which contains the basic
+network model and implementation. The `__mapper_args__` assignment at the
+top is for differentiating this network from the rest. Usually the safe
+thing is to use the same as the subclass, to avoid confusion.
+
+Most simple networks will only need to override the `add_node` method. This
+method is called after a node is added, with the added node as a parameter.
+This method then can decide how and when to connect this node to other nodes
+in the network.
+
+In our code, we first get all nodes in the network (except the new one). If
+the new node is a source, we raise an exception, because due to the circular
+nature of our network, there can be no sources (they don't accept `from`
+connections and can only transmit).
+
+After that, we take the most recent node and connect it to the new node. At
+this point, this is almost the same as a chain network, but when we get to
+the last node, we connect the new node to the first node, in addition to its
+connection to the previous node.
+
+The code in the `add_node` method can be as complex as needed, so very
+complex networks are possible. In most cases, to create a more advanced
+network it will be necessary to add custom properties to it. This is done by
+overriding the `__init__` method of the network to add the properties. The
+following example shows how to do that:
+
+::
+
+    def __init__(self, new_property1, new_property2):
+        self.property1 = repr(new_property1)
```

Comment:
Why are you using repr here?

Reply:
That's because the property values has to be a string (they are defined that way in the SQLAlchemy model). I will add an explanation.